editor/script_create_dialog.cpp

```diff
@@ -145,9 +145,13 @@ void ScriptCreateDialog::ok_pressed() {
 
 void ScriptCreateDialog::_create_new() {
 
-	String cname;
-	if (has_named_classes)
-		cname = class_name->get_text();
+	String cname_param;
+
+	if (has_named_classes) {
+		cname_param = class_name->get_text();
+	} else {
+		cname_param = ProjectSettings::get_singleton()->localize_path(file_path->get_text());
```

Comment:
Shouldn't you use `get_file().get_basename()` here?

Reply:
I wanted to leave the choice to the language, but I guess there is no point.

Reply:
I'd go for what's intuitive for now (class name == file basename), and if the need arises for a language to compose class names based on the full path (like some Java reverse DNS naming maybe?) we can change it at that time.